File: angle-chan-server/src/main/java/com/ben/dektor/store/MediaStore.java
package com.ben.dektor.store;

import com.ben.dektor.entities.MediaInfo;
import org.springframework.content.commons.repository.ContentStore;
import org.springframework.content.rest.StoreRestResource;
import org.springframework.stereotype.Service;

@Service
@StoreRestResource(path = "media")
public interface MediaStore extends ContentStore<MediaInfo, String> {
}

File: angle-chan-server/src/main/kotlin/com/ben/dektor/store/MediaStore.kt
package com.ben.dektor.store

import com.ben.dektor.entities.MediaInfo
import org.springframework.content.commons.repository.ContentStore
import org.springframework.content.rest.StoreRestResource
import org.springframework.stereotype.Component
import javax.annotation.Resource

//@Service
@StoreRestResource(path = "media")
@Component
//@Resource(name = "mediaStore")
interface MediaStore : ContentStore<MediaInfo?, String?>